include/aspect/parameters.h

```diff
@@ -202,6 +202,109 @@ namespace aspect
      */
 
     /**
+     * @name Formulation settings
+     * @{
+     */
+    struct CombinedFormulation
+    {
+      enum Kind
+      {
+        BA,
+        ALA,
+        isothermal_compression,
+        custom
+      };
+      static Kind parse(const std::string &input)
+      {
+        if (input == "isothermal compression")
+          return CombinedFormulation::isothermal_compression;
+        else if (input == "ALA")
+          return CombinedFormulation::ALA;
+        else if (input == "BA")
+          return CombinedFormulation::BA;
+        else if (input == "custom")
+          return CombinedFormulation::custom;
+        else
+          AssertThrow(false, ExcNotImplemented());
+        return CombinedFormulation::isothermal_compression;
+      }
+    };
+
+    struct FormulationMassConservation
+    {
+      enum Kind
+      {
+        isothermal_compression,
+        reference_density_profile,
+        implicit_reference_density_profile,
+        incompressible,
+        ask_material_model
+      };
+      static Kind parse(const std::string &input)
+      {
+        if (input == "isothermal compression")
+          return FormulationMassConservation::isothermal_compression;
+        else if (input == "reference density profile")
+          return FormulationMassConservation::reference_density_profile;
+        else if (input == "implicit reference density profile")
+          return FormulationMassConservation::implicit_reference_density_profile;
+        else if (input == "incompressible")
+          return FormulationMassConservation::incompressible;
+        else if (input == "ask material model")
+          return FormulationMassConservation::ask_material_model;
+        else
+          AssertThrow(false, ExcNotImplemented());
+        return FormulationMassConservation::isothermal_compression;
+      }
+    };
+
+    struct FormulationTemperatureEquation
+    {
+      enum Kind
+      {
+        real_density,
+        reference_density_profile
+      };
+      static Kind parse(const std::string &input)
+      {
+        if (input == "real density")
+          return FormulationTemperatureEquation::real_density;
+        else if (input == "reference density profile")
+          return FormulationTemperatureEquation::reference_density_profile;
+        else
+          AssertThrow(false, ExcNotImplemented());
+        return FormulationTemperatureEquation::real_density;
+      }
+    };
+
+    /**
+     * A consistent way to formulate the equations to solve. Common formulations
+     * are the Boussinesq or Anelastic Liquid Approximations (BA, ALA). ASPECT's
+     * original formulation is termed 'isothermal compression'. 'Custom' allows
+     * to set the approximations individually.
+     */
+    typename CombinedFormulation::Kind combined_formulation;
```

Comment:
The comment has a kind of weird start: "A consistent way to formulate the equations to solve." This doesn't reference what the variable actually does. Maybe "This variable selects one of the consistent ways to formulate..."